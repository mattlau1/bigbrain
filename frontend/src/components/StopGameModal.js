import React from 'react'
import { Button, Col, Container, Modal, Row } from 'react-bootstrap'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const StopGameModal = ({ show, handleClose, id }) => {
  return (
      <Modal show={show} onHide={handleClose}>
      <Modal.Header closeButton>
      <Modal.Title>Game Completed</Modal.Title>
      </Modal.Header>
      <Modal.Body>
      <h4 className="text-center">Would you like to view the results?</h4>
      <Container>
        <Row>
          <Col md={6}>
            <Link to={`/results/${id}`}>
              <Button className='mx-0 w-100 my-1' id='yesClick'>Yes</Button>
            </Link>
          </Col>
          <Col md={6}>
            <Button
              variant="danger"
              className="mx-0 w-100 my-1"
              onClick={() => { handleClose() }}
              id='noClick'
            >
              No
            </Button>
          </Col>
        </Row>
      </Container>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={handleClose} id='closeButton'>
          Close
        </Button>
      </Modal.Footer>
    </Modal>
  )
}

StopGameModal.propTypes = {
  setShow: PropTypes.func,
  handleShow: PropTypes.func,
  handleClose: PropTypes.func,
  show: PropTypes.bool,
  gameList: PropTypes.array,
  setGameList: PropTypes.func,
  id: PropTypes.number
};

export default StopGameModal
