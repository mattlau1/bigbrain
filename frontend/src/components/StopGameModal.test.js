import { shallow } from 'enzyme';
import React from 'react';
import StopGameModal from './StopGameModal';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Stop Game Modal', () => {
  it('should render stop game modal', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  })

  it('should have the question text', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    const text = wrapper.find('h4');
    expect(text.text()).toBe('Would you like to view the results?');
    expect(tree).toMatchSnapshot();
  })

  it('should have the yes button', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    const text = wrapper.find('#view-results-btn');
    expect(text.text()).toBe('Yes');
    expect(tree).toMatchSnapshot();
  })

  it('should have the no button', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    const text = wrapper.find('#no-click');
    expect(text.text()).toBe('No');
    expect(tree).toMatchSnapshot();
  })

  it('should have the close button', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    const text = wrapper.find('#close-button');
    expect(text.text()).toBe('Close');
    expect(tree).toMatchSnapshot();
  })

  it('should not appear without props', () => {
    const tree = renderer.create(<Router><StopGameModal /></Router>).toJSON();
    const wrapper = shallow(<StopGameModal />);
    const button = wrapper.find('#close-button');
    button.simulate('click');
    const modal = shallow(<StopGameModal />);
    expect(modal).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  })
})
