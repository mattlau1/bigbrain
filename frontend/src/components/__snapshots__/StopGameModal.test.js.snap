// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop Game Modal should have the close button 1`] = `null`;

exports[`Stop Game Modal should have the no button 1`] = `null`;

exports[`Stop Game Modal should have the question text 1`] = `null`;

exports[`Stop Game Modal should have the yes button 1`] = `null`;

exports[`Stop Game Modal should not appear without props 1`] = `null`;

exports[`Stop Game Modal should render stop game modal 1`] = `null`;
